File: android_partner_app/app/src/main/java/io/orbitlogic/partner/ui/earnings/EarningsScreen.kt
package io.orbitlogic.partner.ui.earnings

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccountBalance
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.orbitlogic.partner.data.models.PartnerProfile
import io.orbitlogic.partner.data.models.PartnerTransaction
import io.orbitlogic.partner.ui.theme.*

@Composable
fun EarningsScreen(
    profile: PartnerProfile,
    transactions: List<PartnerTransaction>,
    onWithdrawalRequest: (Double) -> Unit,
    onBack: () -> Unit
) {
    var amountText by remember { mutableStateOf("") }
    var isSubmitting by remember { mutableStateOf(false) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(DarkBackground)
            .padding(16.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            IconButton(onClick = onBack) {
                Icon(imageVector = Icons.Default.ArrowBack, contentDescription = null, tint = Color.White)
            }
            Text("Earning Details", color = Color.White, fontSize = 16.sp, fontWeight = FontWeight.W900)
            Spacer(modifier = Modifier.width(48.dp))
        }

        Spacer(modifier = Modifier.height(16.dp))

        // Wallet Card (matches HSL styling layout!)
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    brush = Brush.linearGradient(listOf(PartnerPurple.copy(alpha = 0.15f), Color.Black)),
                    shape = RoundedCornerShape(20.dp)
                )
                .border(1.dp, BorderColor, RoundedCornerShape(20.dp))
                .padding(24.dp)
        ) {
            Column {
                Text("AVAILABLE WALLET BALANCE", color = TextSecondary, fontSize = 10.sp, fontWeight = FontWeight.Bold, letterSpacing = 1.sp)
                Spacer(modifier = Modifier.height(8.dp))
                Text("₹${profile.wallet.balance.toInt()}", color = Color.White, fontSize = 36.sp, fontWeight = FontWeight.W900)
                Spacer(modifier = Modifier.height(20.dp))
                
                // Quick Withdraw form
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    Button(
                        onClick = {
                            if (profile.wallet.balance >= 700.0) {
                                onWithdrawalRequest(profile.wallet.balance)
                            }
                        },
                        colors = ButtonDefaults.buttonColors(containerColor = Color.White),
                        modifier = Modifier.fillMaxWidth().height(48.dp),
                        shape = RoundedCornerShape(12.dp)
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(imageVector = Icons.Default.AccountBalance, contentDescription = null, tint = Color.Black)
                            Spacer(modifier = Modifier.width(8.dp))
                            Text("Withdraw to Bank", color = Color.Black, fontWeight = FontWeight.Bold)
                        }
                    }
                }
            }
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Stats grid
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            StatsCard(label = "Pending Clearance", value = "₹${profile.wallet.pendingClearance.toInt()}", modifier = Modifier.weight(1f))
            StatsCard(label = "Total Withdrawn", value = "₹${profile.wallet.totalWithdrawn.toInt()}", modifier = Modifier.weight(1f))
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Payout history list
        Text("Payout History", color = Color.White, fontSize = 14.sp, fontWeight = FontWeight.Bold)
        Spacer(modifier = Modifier.height(12.dp))

        LazyColumn(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            items(transactions) { tx ->
                TransactionItem(tx)
            }
        }
    }
}

@Composable
fun StatsCard(label: String, value: String, modifier: Modifier = Modifier) {
    Box(
        modifier = modifier
            .background(DarkSurface, RoundedCornerShape(16.dp))
            .border(1.dp, BorderColor, RoundedCornerShape(16.dp))
            .padding(16.dp)
    ) {
        Column {
            Text(label, color = TextSecondary, fontSize = 10.sp)
            Spacer(modifier = Modifier.height(6.dp))
            Text(value, color = Color.White, fontSize = 18.sp, fontWeight = FontWeight.W900)
        }
    }
}

@Composable
fun TransactionItem(tx: PartnerTransaction) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(DarkSurface, RoundedCornerShape(12.dp))
            .border(1.dp, BorderColor, RoundedCornerShape(12.dp))
            .padding(16.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column {
                Text(tx.description, color = Color.White, fontSize = 13.sp, fontWeight = FontWeight.Bold)
                Spacer(modifier = Modifier.height(4.dp))
                Text(tx.createdAt, color = TextSecondary, fontSize = 10.sp)
            }
            Text(
                text = if (tx.type == "EARNING") "+₹${tx.amount.toInt()}" else "-₹${tx.amount.toInt()}",
                color = if (tx.type == "EARNING") Color(0xFF2ECC71) else Color(0xFFE74C3C),
                fontSize = 14.sp,
                fontWeight = FontWeight.W900
            )
        }
    }
}

File: partner_app/lib/features/earnings/earnings_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../core/theme.dart';
import '../../core/api_service.dart';
import '../home/home_screen.dart';
import '../../models/partner_profile.dart';

class EarningsScreen extends ConsumerStatefulWidget {
  const EarningsScreen({super.key});

  @override
  ConsumerState<EarningsScreen> createState() => _EarningsScreenState();
}

class _EarningsScreenState extends ConsumerState<EarningsScreen> {
  bool _isWithdrawing = false;

  void _withdrawFunds() async {
    final profile = ref.read(partnerProfileProvider);
    if (profile == null) return;

    if (profile.wallet.balance <= 0) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text('No available balance to withdraw.')),
      );
      return;
    }

    setState(() => _isWithdrawing = true);
    final api = ApiService();
    final success = await api.withdrawWallet(profile.id, profile.wallet.balance);
    setState(() => _isWithdrawing = false);

    if (success && mounted) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text('🎉 Withdrawal request submitted! Funds will clear in 24 hours.'), backgroundColor: Colors.green),
      );
      
      // Update local wallet balance
      ref.read(partnerProfileProvider.notifier).state = profile.copyWith(
        wallet: PartnerWallet(
          balance: 0.0,
          pendingClearance: profile.wallet.pendingClearance,
          totalWithdrawn: profile.wallet.totalWithdrawn + profile.wallet.balance,
        ),
      );
    }
  }

  @override
  Widget build(BuildContext context) {
    final profile = ref.watch(partnerProfileProvider);
    if (profile == null) return const Center(child: CircularProgressIndicator());

    final balance = profile.wallet.balance;
    final pending = profile.wallet.pendingClearance;
    final withdrawn = profile.wallet.totalWithdrawn;

    return ListView(
      padding: const EdgeInsets.all(16.0),
      children: [
        const Text('Earning Details', style: TextStyle(fontSize: 18, fontWeight: FontWeight.w900)),
        const SizedBox(height: 16),

        // Wallet Balance Card (matches design images!)
        Container(
          padding: const EdgeInsets.all(24),
          decoration: BoxDecoration(
            gradient: LinearGradient(colors: [OrbitTheme.partnerPurple.withValues(alpha: 0.15), Colors.black]),
            borderRadius: BorderRadius.circular(20),
            border: Border.all(color: OrbitTheme.border),
          ),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Text('AVAILABLE WALLET BALANCE', style: TextStyle(color: OrbitTheme.textSecondary, fontSize: 10, fontWeight: FontWeight.bold, letterSpacing: 1)),
              const SizedBox(height: 8),
              Text(
                '₹${balance.toInt()}',
                style: const TextStyle(fontSize: 36, fontWeight: FontWeight.w900, color: Colors.white),
              ),
              const SizedBox(height: 20),
              SizedBox(
                width: double.infinity,
                height: 48,
                child: ElevatedButton.icon(
                  style: ElevatedButton.styleFrom(
                    backgroundColor: Colors.white,
                    foregroundColor: Colors.black,
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
                  ),
                  icon: const Icon(Icons.arrow_outward),
                  label: _isWithdrawing
                      ? const SizedBox(width: 20, height: 20, child: CircularProgressIndicator(strokeWidth: 2, color: Colors.black))
                      : const Text('Withdraw Payouts', style: TextStyle(fontWeight: FontWeight.bold)),
                  onPressed: _isWithdrawing ? null : _withdrawFunds,
                ),
              ),
            ],
          ),
        ),

        const SizedBox(height: 16),

        // Double statistics row
        Row(
          children: [
            Expanded(
              child: _buildStatCard('Pending Clearance', '₹${pending.toInt()}', Colors.amberAccent),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: _buildStatCard('Total Withdrawn', '₹${withdrawn.toInt()}', Colors.greenAccent),
            ),
          ],
        ),

        const SizedBox(height: 28),
        const Text('Shoot Transactions Log', style: TextStyle(fontSize: 15, fontWeight: FontWeight.bold)),
        const SizedBox(height: 12),

        // Transaction History Items showing flat ₹700 salary per completed order
        _buildTransactionItem('Shoot Payout (Noida Sector 62)', '₹700', '28 June 2026', true),
        _buildTransactionItem('Shoot Payout (Noida Sector 18)', '₹700', '25 June 2026', true),
        _buildTransactionItem('Wallet Cash Withdrawal', '-₹2100', '20 June 2026', false),
      ],
    );
  }

  Widget _buildStatCard(String label, String value, Color color) {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: OrbitTheme.cardBackground,
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: OrbitTheme.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(label, style: const TextStyle(fontSize: 10, color: OrbitTheme.textSecondary)),
          const SizedBox(height: 6),
          Text(value, style: TextStyle(fontSize: 18, fontWeight: FontWeight.w900, color: color)),
        ],
      ),
    );
  }

  Widget _buildTransactionItem(String title, String amount, String date, bool isEarning) {
    return Container(
      margin: const EdgeInsets.only(bottom: 8),
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: OrbitTheme.cardBackground,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: OrbitTheme.border),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(title, style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 13)),
              const SizedBox(height: 4),
              Text(date, style: const TextStyle(color: OrbitTheme.textSecondary, fontSize: 10)),
            ],
          ),
          Text(
            amount,
            style: TextStyle(
              fontWeight: FontWeight.w900,
              fontSize: 14,
              color: isEarning ? Colors.greenAccent : Colors.redAccent,
            ),
          ),
        ],
      ),
    );
  }
}
